src/vm/lib/metadata/agent.js

```diff
@@ -1121,7 +1177,20 @@ function createZoneSocket(zopts, callback) {
             zlog.info('listening on fd %d', fd);
             self.addDebug(zopts.zone, 'last_zsock_listen_success');
 
-            callback();
+            fs.stat(self.zoneConnections[zopts.zone].sockpath,
```

Comment:
Is there a way to not duplicate that code with `createKVMServer`'s implementation?